src/Symfony/Component/Translation/Tests/Command/TranslationPullCommandTest.php

```diff
@@ -720,16 +722,55 @@ public static function provideCompletionSuggestions(): \Generator
         ];
     }
 
-    private function createCommandTester(ProviderInterface $provider, array $locales = ['en'], array $domains = ['messages'], $defaultLocale = 'en'): CommandTester
+    /**
+     * @dataProvider provideEventDispatchingCommands
+     */
+    public function testEventIsDispatched(array $command)
+    {
+        $dispatcher = $this->createMock(EventDispatcherInterface::class);
+        $dispatcher->expects(self::once())
+            ->method('dispatch')->with(self::callback(static fn (TranslationPullEvent $event): bool => true));
+
+        $providerReadTranslatorBag = new TranslatorBag();
+
+        $provider = $this->createMock(ProviderInterface::class);
+        $provider->expects($this->once())
+            ->method('read')
+            ->willReturn($providerReadTranslatorBag);
+
+        $tester = $this->createCommandTester(provider: $provider, dispatcher: $dispatcher);
+
+        $tester->execute($command);
+    }
+
+    public static function provideEventDispatchingCommands(): \Generator
+    {
+        yield 'without force' => [
+            'command' => [
+                '--locales' => ['en'],
+                '--domains' => ['messages'],
+            ],
+        ];
+
+        yield 'with force' => [
+            'command' => [
+                '--locales' => ['en'],
+                '--domains' => ['messages'],
+                '--force' => '',
+            ],
+        ];
+    }
+
+    private function createCommandTester(ProviderInterface $provider, array $locales = ['en'], array $domains = ['messages'], $defaultLocale = 'en', EventDispatcherInterface $dispatcher = null): CommandTester
     {
-        $command = $this->createCommand($provider, $locales, $domains, $defaultLocale);
+        $command = $this->createCommand(provider: $provider, locales: $locales, domains: $domains, defaultLocale: $defaultLocale, dispatcher: $dispatcher);
```

Comment:
When using named arguments, I propose to use line breaks
```suggestion
        $command = $this->createCommand(
            provider: $provider,
            locales: $locales,
            domains: $domains,
            defaultLocale: $defaultLocale,
            dispatcher: $dispatcher
        );
```